k8s/controllers/fabric/target_controller.go

```diff
@@ -93,13 +95,49 @@ func (r *TargetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	deploymentOperationType := metrics.DeploymentQueued
 	var err error
 
+	version := target.Spec.Version
+	name := target.Spec.RootResource
+	targetName := name + ":" + version
+	jData, _ := json.Marshal(target)
+
 	if target.ObjectMeta.DeletionTimestamp.IsZero() { // update
+		_, exists := target.Labels["version"]
+		log.Info(fmt.Sprintf("Target update: version tag exists - %v", exists))
+		if !exists && version != "" && name != "" {
+			err := r.ApiClient.CreateTarget(ctx, targetName, jData, req.Namespace, "", "")
```

Comment:
I'm very concerned about this. We are calling the creation API inside the controller. The API calls may lead to a K8s state provider, which triggers reconciliation again. Also, we are rewriting the object key in reconciliation logic. I'm certain this is NOT an anticipated behavior of Kubebuilder users nor K8s APIs in general, hence I'm not sure what hidden consequences this may lead to.

Reply:
Updating metdata.labels doesn't make generation change.  To avoid the second reconcilication triggered, add `WithEventFilter(predicate.GenerationChangedPredicate{}).` in `SetupWithManager`. Do you have concerns about adding the event filter?